File: backend/src/main/java/com/bookloop/api/book/Book.java
package com.bookloop.api.book;

import com.bookloop.api.book.photo.BookPhoto;
import com.bookloop.api.user.User;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

import java.util.ArrayList;
import java.util.List;

@Entity
@Data
@Table(name = "books")
public class Book {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank
    private String title;

    @NotBlank
    private String author;

    private String isbn;

    @NotBlank
    private String description;

    @Enumerated(EnumType.STRING)
    @NotNull
    private BookStatus status;

    @NotNull
    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @OneToMany(mappedBy = "book", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<BookPhoto> photos = new ArrayList<>();

}

File: mobile/models/Book.ts
import { BookPhoto } from "./BookPhoto";

export interface Book {
  id?: number;

  title: string;
  author: string;
  isbn?: string;
  description: string;

  status: "DOACAO" | "TROCA" | "INATIVO";

  condition: "NOVO" | "SEMINOVO" | "BOM" | "REGULAR" | "RUIM";

  userId?: number;
  userName?: string;

  photos: BookPhoto[];
}